index.js

```diff
@@ -190,11 +191,16 @@ module.exports = {
               entry => entry.__frameId === frameId
             );
 
-            if (rootFrameMappings.has(frameId) || previousFrameId) {
+            if (rootFrameMappings.has(frameId)) {
               // This is a sub frame, there's already a page for the root frame
               continue;
             }
 
+            if (options.splitPagesWithinFrame === false && previousFrameId) {
```

Comment:
options.splitPagesWithinFrame !== true would be better here, otherwise true, NaN, undefined and anything that's not false would be treated the same.